components/kyma-metrics-collector/pkg/process/process.go

```diff
@@ -353,25 +353,59 @@ func isSuccess(status int) bool {
 	return false
 }
 
-func isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
-	if runtime.Status.Provisioning != nil &&
-		runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
-		runtime.Status.Deprovisioning == nil {
+// Helper function which returns true if the runtime state is trackable
+func isTrackableState(state kebruntime.State) bool {
+	switch state {
+	case kebruntime.StateSucceeded, kebruntime.StateError, kebruntime.StateUpgrading, kebruntime.StateUpdating:
 		return true
 	}
 	return false
 }
 
+func (p *Process) isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
+	// Check if the cluster is in a trackable state
+	if isTrackableState(runtime.Status.State) ||
+		(runtime.Status.Provisioning != nil &&
+			runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
+			runtime.Status.Deprovisioning == nil) {
+
+		p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).With(log.KeyRuntimeID, runtime.RuntimeID).
+			Infof("Tracked cluster of state: %s, provisioning state: %s, no deprovisioning state.",
+				runtime.Status.State, runtime.Status.Provisioning.State)
+		return true
+	}
+
+	// Set provisioning and deprovisioning states if available
+	provisioning := getOrDefault(runtime.Status.Provisioning, "")
+	deprovisioning := getOrDefault(runtime.Status.Deprovisioning, "")
+
+	p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).With(log.KeyRuntimeID, runtime.RuntimeID).
+		Infof("Not tracking cluster of state: %s, provisioning state: %s, deprovisioning state: %s",
+			runtime.Status.State, provisioning, deprovisioning)
```

Comment:
Please apply the same logging pattern described [here](https://github.com/kyma-project/control-plane/pull/3334/files#r1500365859).